### ISSUE TITLE
[FEATURE] CCM support resync nodes topology after migrated/updated nodes

### ISSUE THREAD

**Issue by chrisho** [harvester/harvester#3142]
**Is your feature request related to a problem? Please describe.**

Since ccm API limitation, ccm doesn't support resync nodes topology after migrated or updated nodes.

**Describe the solution you'd like**

resync nodes topology after migrated or updated nodes.

**Describe alternatives you've considered**

**Additional context**

**Comment by guangbochen** [comment]
We will postpone the CCM version bump in both RKE2 and RKE1-chart on the June release, so this issue can be verified first once the new CCM chart is ready, but the official release would depends on the upstream RKE2 & KDM releases.

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.2-27934bc3-head` (06/07) and Rancher `v2.7.5-rc4`. Close this issue

### Result
When we have different labels set on each node for the following:
1. node1: `topology.kubernetes.io/region=region1` and `topology.kubernetes.io/zone=zone1`
2. node2: `topology.kubernetes.io/region=region2` and `topology.kubernetes.io/zone=zone2` 
3. node3: `topology.kubernetes.io/region=region3` and `topology.kubernetes.io/zone=zone3` 

**After we migrate guest cluster vm to each node, the nodes topology label will be synced according to the target node**

* Before migrate, we can see in the Rancher `Nodes` page, the guest cluster vm located on `node3` have labels `topology.kubernetes.io/region=region3` and `topology.kubernetes.io/zone=zone3` 
  ![image](https://github.com/harvester/harvester/assets/29251855/ecb4d6ce-0443-4a9a-b8e4-4d562bca2fd1)


* We migrate vm to `node1` in Harvester
  ![image](https://github.com/harvester/harvester/assets/29251855/af5d3311-40c8-4a9b-b595-793b5d932997)

* After guest cluster migrated to `node1`, the lables have been updated to `topology.kubernetes.io/region=region1` and `topology.kubernetes.io/zone=zone1`
  ![image](https://github.com/harvester/harvester/assets/29251855/9bf0183c-37d3-43fc-b4d4-9327cbf8b8ab)

* When we update or clean up the labels on `node2` to `topology.kubernetes.io/region=region_none` and `topology.kubernetes.io/zone=zone_none` and migrate vm to `node2` 

* the lables have been updated to `topology.kubernetes.io/region=region_none` and `topology.kubernetes.io/zone=zone_none`

  ![image](https://github.com/harvester/harvester/assets/29251855/d3889fc4-db3d-433b-b7c2-c1ccb21d469d)

### Test Information
* Test Environment: 3 nodes harvester on bare machines
* Harvester version: `v1.2-27934bc3-head` (06/07)
* Rancher `v2.7.5-rc4`

### Verify Steps
Ref: https://github.com/harvester/cloud-provider-harvester/pull/30#issue-1531664462 

1. Add labels topology.kubernetes.io/zone or topology.kubernetes.io/region for per harvester node.
2. Create a rke2 guest cluster on Rancher.
3. Wait for the guest cluster to be completed.
4. ssh to the rke2 guest cluster VM
5. $ cd `/var/lib/rancher/rke2/bin`
6. Install helm 
7. Add harvester repo 
`$ helm repo add harvester https://charts.harvesterhci.io`
8. update repo 
`$ helm repo update`
9. Uninstall current ccm chart 
`$ helm uninstall -n kube-system harvester-cloud-provider`
10. `export KUBECONFIG=/etc/rancher/rke2/rke2.yaml`
11. Install the new ccm chart 
`$ helm install harvester-cloud-provider harvester/harvester-cloud-provider -n kube-system --set cloudConfigPath=/var/lib/rancher/rke2/etc/config-files/cloud-provider-config` 
    ```
    NAME: harvester-cloud-provider
    LAST DEPLOYED: Fri Jun  9 03:57:50 2023
    NAMESPACE: kube-system
    STATUS: deployed
    REVISION: 1
    TEST SUITE: None
    ```
12. Edit cloud-provider deployment 
`$ ./kubectl edit deployments -n kube-system harvester-cloud-provider`
13. change the cloud-provider image to `0.2.0`
  ```
  containers:
        - args:
          - --cloud-config=/etc/kubernetes/cloud-config
          command:
          - harvester-cloud-provider
          image: rancher/harvester-cloud-provider:v0.2.0
  ```
14. Check the cloud-provider pod version have been updated to `0.2.0`
![image](https://github.com/harvester/harvester/assets/29251855/19713cb6-3506-4ced-a670-bf3a6dac2bfd)

15.  Migrate guest cluster vm from `node3` to `node1`
16. Open RKE2 guest cluster -> Nodes, check the topology label have been updated according to `node1`
17. Edit and update the node2 topology labels values
18.  Migrate guest cluster vm from `node1` to `node2`
19. Open RKE2 guest cluster -> Nodes, check the topology label have been updated according to `node2`

#### Tested the previous steps also works on the embedded Rancher
